overlord/snapstate/handlers.go

```diff
@@ -642,6 +643,28 @@ func hasOtherInstances(st *state.State, instanceName string) (bool, error) {
 	return false, nil
 }
 
+var ErrKernelGadgetUpdateTaskMissing = errors.New("cannot refresh kernel without a gadget update task")
+
+func checkKernelHasUpdateAssetsTask(t *state.Task) error {
```

Comment:
We should probably also add an "assumes" for this feature and add it to the kernel(?)

Reply:
well either that or we could add to the kernel `assumes: [snapd2.51.7]` or something now that assumes works with sub-versions like that

Reply:
I'm probably missing something but would the assume prevent? it's a while that snapd does the right thing

Reply:
well I was thinking that having an assumes would prevent unnecessary churn where devices running snapd without this fix would not try to refresh the kernel, but I suppose the fact that snapd in the field doesn't handle assumes early enough means that adding a new assumes to the kernel before an updated snapd is available would be ineffective